src/linking_number_certificate.cpp

```diff
@@ -357,15 +357,22 @@ void ComputeLinkingNumbersDirectSum(
  */
 void LinkingNumberCertificate::ComputeFromModel(const Model &model,
                                                 bool force_direct_sum,
+                                                bool discretize,
                                                 double barnes_hut_init_beta,
                                                 double barnes_hut_beta_limit) {
   std::cout << "Starting Potential Link Search." << std::endl;
```

Comment:
This code is a bit tricky --- the difference between `potential_links` and `potential_links_unique_per_curve` is the former has each pair twice and the latter has each pair appear once (e.g. "(2,1)" appears also as "(1,2)" in potential_links). The duplication is just there to help with parallelizing discretization.

So this is almost there, but the linking number computation will end up doing double the work and produce a full instead of a triangular matrix.

It would be great if you can
1) move the `PotentialLinkSearch` step into the `if (discretize)` section (because it duplicates the pairs), and
2) In the else section, directly perform a unique-list potential link search

So the code starting from this line (363) should end up looking like this:
```
  std::cout << "Starting Potential Link Search." << std::endl;
  std::vector<Curve> discretized_curves;
  std::vector<std::vector<int>> potential_links_unique_per_curve;
  if (discretize) {
    std::vector<std::vector<int>> potential_links = PotentialLinkSearch(model);
    std::cout << "Starting Discretization." << std::endl;
    DiscretizeAndGetNewPotentialLinks(model, potential_links,
                                      discretized_curves,
                                      potential_links_unique_per_curve);
  } else {
    // Perform PLS where each pair appears only once, and then skip
    // discretization.
    PrecomputeCurveBoundingBoxes(model.GetCurves());
    potential_links_unique_per_curve =
        GetPotentialLinksUniqueListPerCurve(model.GetCurves());
    std::cout << "Skipping Discretization." << std::endl;
    discretized_curves = model.GetCurves();
  }
```